pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -228,24 +231,40 @@ async def _initialize_context(self):
     def _register_to_signal(self):
         on_termination_signal = functools.partial(self._on_termination_signal, Constants.termination_signal.name)
         on_drain_signal = functools.partial(self._on_drain_signal, Constants.drain_signal.name)
+        on_continue_signal = functools.partial(self._on_continue_signal, Constants.continue_signal.name)
 
         asyncio.get_running_loop().add_signal_handler(Constants.termination_signal, on_termination_signal)
         asyncio.get_running_loop().add_signal_handler(Constants.drain_signal, on_drain_signal)
+        asyncio.get_running_loop().add_signal_handler(Constants.continue_signal, on_continue_signal)
 
     def _on_drain_signal(self, signal_name):
-        self._logger.debug_with('Received signal, calling draining callback', signal=signal_name)
+        # do not perform draining if discarding events
+        if self._discard_events:
+            return
+
+        self._logger.debug_with('Received signal', signal=signal_name)
         self._is_drain_needed = True
+
+        # set the flag to True to stop processing events which are received after draining
+        self._discard_events = True
+
         # if serving loop is waiting for an event, unblock this operation to allow the drain callback to be called
         if self._event_message_length_task:
             self._event_message_length_task.cancel()
 
     def _on_termination_signal(self, signal_name):
-        self._logger.debug_with('Received signal, calling termination callback', signal=signal_name)
+        self._logger.debug_with('Received signal', signal=signal_name)
         self._is_termination_needed = True
         # if serving loop is waiting for an event, unblock this operation to allow the termination callback to be called
         if self._event_message_length_task:
             self._event_message_length_task.cancel()
 
+    def _on_continue_signal(self, signal_name):
+        self._logger.debug_with('Received continue signal', signal=signal_name)
```

Comment:
`self._logger.debug_with('Received signal', signal=signal_name)`
for consistency with `_on_drain_signal`  and `_on_termination_signal`